tests/unit_tests.cpp

```diff
@@ -0,0 +1,133 @@
+#include "./lib/catch.hpp"
+#include "choices.hpp"
+#include "feature.hpp"
+#include "model.hpp"
+#include "polymer.hpp"
+#include "reaction.hpp"
+#include "tracker.hpp"
+
+TEST_CASE("Genome construction")
+{
+    auto plasmid = std::shared_ptr<Genome>(new Genome("T7", 305, 0.01, 0.001, 9.0, 20.0));
+    auto plasmid2 = std::shared_ptr<Genome>(new Genome("T7", 305));
+    
+    REQUIRE(plasmid->transcript_degradation_rate() == 0.01);
+    REQUIRE(plasmid->transcript_degradation_rate_ext() == 0.001);
+    REQUIRE(plasmid->rnase_speed() == 9);
+    REQUIRE(plasmid->rnase_footprint() == 20);
+
+    //Test the genome default constructor
+    REQUIRE(plasmid2->transcript_degradation_rate() == 0.0);
+    REQUIRE(plasmid2->transcript_degradation_rate_ext() == 0.0);
+    REQUIRE(plasmid2->rnase_speed() == 0.0);
+    REQUIRE(plasmid2->rnase_footprint() == 0);
+}
+
+TEST_CASE("Add mask to genome")
+{
+    auto plasmid = std::shared_ptr<Genome>(new Genome("T7", 305));
+    std::vector<std::string> interactions = {"rnapol","ecolipol"};
+    plasmid->AddMask(15,interactions);
+
+    REQUIRE(plasmid->GetMask().start() == 15);
+    REQUIRE(plasmid->GetMask().stop() == 305);
+    REQUIRE(plasmid->GetMask().CheckInteraction("rnapol"));
+    REQUIRE(plasmid->GetMask().CheckInteraction("ecolipol"));
+}
+
+TEST_CASE("Add fixed elements to a genome")
+{
+    //Create a genome and add a promoter
+    auto plasmid = std::shared_ptr<Genome>(new Genome("T7", 305));
+    std::map<std::string, double> interactions = {{"rnapol",2e8}};
+    plasmid->AddPromoter("phi1",1,10,interactions);
+    CHECK(plasmid->bindings().size() == 1);
+    CHECK(plasmid->GetBindingIntervals().size() == 1 );
+    
+    //Test the AddTerminator method
+    std::map<std::string, double> efficiency = {{"rnapol",1.0}};
+    plasmid->AddTerminator("t1",304,305,efficiency);
+    CHECK(plasmid->GetReleaseIntervals().size() == 1);
+
+    //Add two genes to the genome
+    //There should now be three binding sites ('bindings') -
+    //one polymerase binding site and two ribosome binding sites
+    plasmid->AddGene("rnapol",26,225,11,26,1e7);
+    plasmid->AddGene("proteinX",241,280,226,241,1e7);
+    REQUIRE(plasmid->bindings().size() == 3);
+}
```

Comment:
In this entire function, pay attention to spaces after commas. Also in the rest of the code. I didn't mark every case individually.

Reply:
I think this should be all good now.